code/modules/antagonists/blob/blobstrains/_blobstrain.dm

```diff
@@ -24,16 +62,66 @@ GLOBAL_LIST_INIT(valid_blobstrains, subtypesof(/datum/blobstrain) - list(/datum/
 
 /datum/blobstrain/proc/on_gain()
 	overmind.color = complementary_color
-	for(var/BL in GLOB.blobs)
-		var/obj/structure/blob/B = BL
+
+	if(overmind.blob_core)
+		overmind.blob_core.max_spores += core_spore_bonus
+		overmind.blob_core.claim_range += core_range_bonus
+		overmind.blob_core.pulse_range += core_range_bonus
+		overmind.blob_core.expand_range += core_range_bonus
+		overmind.blob_core.strong_reinforce_range += core_strong_reinforcement_range_bonus
+		overmind.blob_core.reflector_reinforce_range += core_reflector_reinforcement_range_bonus
+
+	for(var/obj/structure/blob/special/node/N in overmind.node_blobs)
```

Comment:
Do these lists contain types different than these ones? If not, use the `as anything` keyword to avoid the unnecessary type-check.